File: raw_materials/Sauvergarde_Patrimoine_SWH/fonct/getbof.f
C*GETBOF
C===============================================================!
C								!
C	Giraudon  -  SOPHIA     :      3/05/84			!
C								!
C===============================================================!
C								!
C	Routine qui calcule le nombre de lignes maximum pouvant !
C	etre lue a la fois, en fonction du format de l'image    !
C	et de la taille du buffer lorsque l'on veut filtrer une !
C	image par bout (probleme du recouvrement)		!
C								!
C	Les parametres d'entrees sont :				!
C	- IFMT   :  format de l'image				!
C	- LONG   :  taille du buffer de stockage		!
C	- IDIM   :  dimension du filtre				!
C								!
C	Les parametres de sorties sont :			!
C	- NLIGN  : nombre de lignes images pouvant etre lues	!
C		   a la fois					!
C	- NREST  : nombre de lignes a lire au dernier coup	!
C	- NTOUR  : nombre de tour pour lire toute l'image       !
C	- NDIMA  : le nombre exact de lignes utiles		!
C								!
C===============================================================!
C-
	subroutine getbof(ifmt,long,idim,nlign,nrest,ntour,ndima)
	dimension ifmt(1)
	index=0
	inf=1
	nlign=long/ifmt(1)
	max=nlign
	do 5 i=1,600
	index=index+1
	if (max.ge.ifmt(2)) goto 200
	inf=max-idim+2
	max=inf+nlign-1
5	continue
200	nrest=nlign-(max-ifmt(2))
	if (nlign.ge.ifmt(2)) nlign=ifmt(2)
	ntour=index
	ndima=nlign-idim+1
	return
	end

File: fonct/getbof.c
/**GETBOF */
/*===============================================================*/
/*								*/
/*	Giraudon  -  SOPHIA     :      3/05/84			*/
/*								*/
/*===============================================================*/
/*								*/
/*	Routine qui calcule le nombre de lignes maximum pouvant */
/*	etre lue a la fois, en fonction du format de l'image    */
/*	et de la taille du buffer lorsque l'on veut filtrer une */
/*	image par bout (probleme du recouvrement)		*/
/*								*/
/*	Les parametres d'entrees sont :				*/
/*	- IFMT   :  format de l'image				*/
/*	- LONG   :  taille du buffer de stockage		*/
/*	- IDIM   :  dimension du filtre				*/
/*								*/
/*	Les parametres de sorties sont :			*/
/*	- NLIGN  : nombre de lignes images pouvant etre lues	*/
/*		   a la fois					*/
/*	- NREST  : nombre de lignes a lire au dernier coup	*/
/*	- NTOUR  : nombre de tour pour lire toute l'image       */
/*	- NLIT   : le nombre exact de lignes utiles		*/
/*								*/
/*===============================================================*/


getbof(ifmt,ilong,idim,nlign,nrest,ntour,nlit)
int ifmt[], ilong, idim, *nlign, *nrest, *ntour, *nlit ; {

	int index=0,inf=1 , max ,i ;
	*nlign = ilong / ifmt[0] ;
	max = *nlign ;
	for( i = 0 ; i< ifmt[1] ; i++) {
		index = index + 1 ;
		if (max >= ifmt[1]) goto LABEL ;
		inf = max - idim + 2 ;
		max = inf + *nlign - 1 ;
	}

LABEL:	*nrest = *nlign - (max-ifmt[1] ) ;
	if (*nlign >= ifmt[1] ) *nlign=ifmt[1] ;
	*ntour = index ;
	*nlit = *nlign - idim + 1 ;
	return(1) ;

}
